src/main/java/sirius/biz/process/logs/ProcessLog.java

```diff
@@ -398,20 +419,34 @@ public ProcessLog withFormattedMessage(String message, Object... parameters) {
     /**
      * Specifies a handled handledException from where to inherit the message.
      * <p>
-     * If set, the {@link ProcessContext#HINT_MESSAGE_TYPE} hint is used to categorize the log in message types
-     * and {@link ProcessContext#HINT_MESSAGE_COUNT} to limit the amount of allowed messages per type.
+     * If set, the {@link #HINT_MESSAGE_TYPE} or {@link #HINT_MESSAGE_KEY} is used to categorize the log in
+     * message types and {@link #HINT_MESSAGE_COUNT} to limit the amount of allowed messages per type.
      *
      * @param handledException the {@link HandledException} to retrieve the message and eventually hints from
      * @return the log entry itself for fluent method calls
      */
     public ProcessLog withHandledException(HandledException handledException) {
         this.withMessage(handledException.getMessage());
-        handledException.getHint(ProcessContext.HINT_MESSAGE_TYPE).ifFilled(hint -> {
-            int messageCount = handledException.getHint(ProcessContext.HINT_MESSAGE_COUNT).asInt(0);
+        ValueHolder<String> hintMessage = ValueHolder.of(null);
```

Comment:
anit-pattern :-P -> whenever a variable is initialized with "null" => extract method